### PR TITLE
Rename library, API symbols, and switch to Meson

### PR DESCRIPTION
**Pull request by ptomato**
https://phabricator.endlessm.com/T21655

### CODE REVIEW

File: dmodel/dm-macros.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#if DM_VERSION_MAX_ALLOWED < DM_VERSION_MIN_REQUIRED
+#error "DM_VERSION_MAX_ALLOWED must be >= DM_VERSION_MIN_REQUIRED"
+#endif
+#if DM_VERSION_MIN_REQUIRED < DM_VERSION_1_0
+#error "DM_VERSION_MIN_REQUIRED must be >= DM_VERSION_1_0"
+#endif
+
+/* Every new stable minor release should add a set of macros here */
+
+#if DM_VERSION_MIN_REQUIRED >= DM_VERSION_0_0
```
Comment (ebassi) [inline dmodel/dm-macros.h]: We cannot have deprecated API, at this point, so this whole section can be omitted.

File: dmodel/dm-macros.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+#  define DM_VERSION_MAX_ALLOWED (DM_VERSION_CUR_STABLE)
+# endif
+#endif
+
+/* sanity checks */
+#if DM_VERSION_MAX_ALLOWED < DM_VERSION_MIN_REQUIRED
+#error "DM_VERSION_MAX_ALLOWED must be >= DM_VERSION_MIN_REQUIRED"
+#endif
+#if DM_VERSION_MIN_REQUIRED < DM_VERSION_1_0
+#error "DM_VERSION_MIN_REQUIRED must be >= DM_VERSION_1_0"
```
Comment (ebassi) [inline dmodel/dm-macros.h]: The minimum version is 0.0, unless we want to release 1.0.

File: dmodel/dm-macros.h [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+/* Each new cycle should add a new version symbol here */
+
+/**
+ * DM_VERSION_0_0:
+ *
+ * A pre-processor macro that evaluates to the 0.0 version of DModel, in a
+ * format that can be used by %DM_VERSION_MIN_REQUIRED and
+ * %DM_VERSION_MAX_ALLOWED.
+ */
+#define DM_VERSION_0_0 (G_ENCODE_VERSION (0, 0))
```
Comment (ebassi) [inline dmodel/dm-macros.h:45]: I'd honestly drop this: it's kind of pointless to tag at 0.0.

File: dmodel/dm-macros.h [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+#if DM_VERSION_MIN_REQUIRED >= DM_VERSION_0_0
+# define DM_DEPRECATED_IN_0_0        DM_DEPRECATED
+# define DM_DEPRECATED_IN_0_0_FOR(f) DM_DEPRECATED_FOR(f)
+#else
+# define DM_DEPRECATED_IN_0_0        _DM_EXTERN
+# define DM_DEPRECATED_IN_0_0_FOR(f) _DM_EXTERN
+#endif
+
+/* unconditional: we can't have an earlier API version */
+#define DM_AVAILABLE_IN_0_0                  _DM_EXTERN
```
Comment (ebassi) [inline dmodel/dm-macros.h]: At this point, I think I'd drop all of these, and just have a `DM_AVAILABLE_IN_ALL` as the marker for the API introduced from the first release. This way we get to remove the `DM_VERSION_0_0` and every other branching.

Unless, of course, you want to release a 1.0 version now, in which case having `DM_VERSION_1_0` would still make sense, but I'd still use a `DM_AVAILABLE_IN_ALL` annotation.

**Review by ebassi — CHANGES_REQUESTED** [review]

File: dmodel/dm-macros.h (hunk 3 above)
Comment (ptomato) [reply dmodel/dm-macros.h:45]: I can imagine it being useful for defining `DM_VERSION_MAX_ALLOWED` to 0.0. I removed some of the other stuff, but for now I kept it for that reason.

**Comment by ptomato** [comment]
Fixed up the versioning macros.

**Review by ebassi — APPROVED** [review]

**Pull request merged**